lib/ask/runtime/channel_broker.ex

```diff
@@ -355,27 +349,27 @@ defmodule Ask.Runtime.ChannelBroker do
     ChannelBrokerSupervisor.terminate_child(channel_id)
   end
 
-  def handle_info({:collect_garbage, channel_type}, state) do
-    info("handle_info[collect_garbage]", channel_type: channel_type, config: state.config)
+  def handle_info({:collect_garbage}, state) do
+    info("handle_info[collect_garbage]", channel_type: state.channel_type, config: state.config)
 
-    active_respondents =
-      from(r in "respondents",
-        where: r.id in ^Map.keys(state.active_contacts) and r.state == "active",
-        select: r.id
+    active_respondent_ids =
+      Repo.all(
+        from r in "respondents",
+          select: r.id,
+          where: r.id in ^State.active_respondent_ids(state) and r.state == "active"
```

Comment:
This is already limited to the channel capacity (e.g. 200) not the batch size (e.g. 20000) since the query restricts to respondents that the channel believes are active.

Maybe we could reverse the logic to return inactive respondents instead of active ones? We expect there won't be less active than we think there are. Then we could paginate and clean-up the respondents from the channel-broker-queue in multiple passes. :thinking: